gaphor/ui/diagrampage.py

```diff
@@ -157,19 +156,24 @@ def construct(self):
 
         return self.widget
 
-    def get_tool(self, tool_name):
-        """Return a tool associated with an id (action name)."""
-        if tool_name == "toolbox-pointer":
-            return DefaultTool(self.event_manager)
-
-        tool = next(
+    def get_tool_def(self, tool_name):
+        return next(
             t
             for t in tooliter(self.modeling_language.toolbox_definition)
             if t.id == tool_name
         )
-        item_factory = tool.item_factory
-        handle_index = tool.handle_index
-        return PlacementTool(
+
+    def apply_tool_set(self, tool_name):
```

Comment:
I like this refactoring of the toolbox methods :+1:, nice idea!

Reply:
Thanks :blush: